lib/jobs/ucs-service-profile-discovery.js

```diff
@@ -0,0 +1,197 @@
+            })
+            .spread(function() {
+                self._done();
+            })
+            .catch(function(err) {
+                self._done(err);
+            });
+    };
+
+    /**
+     * @function logIn
+     * @description Log into the UCS microservice
+     */
+    UcsServiceProfileDiscoveryJob.prototype.logIn = function () {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var host = this.settings.ucs;
+        var url= "/login?host=" + host+ "&user="+ username + "&password=" + password;
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+
+    /**
+     * @function getRoot
+     * @description return all the elements from the Cisco UCS Manager that RackHD cares about
+     */
+    UcsServiceProfileDiscoveryJob.prototype.getRoot = function () {
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var url=
+            "/serviceProfile?host=" + ucs+ "&user="+ username + "&password=" + password;
+        return this.ucs.clientRequest(url)
+            .then(function(response) {
+                return response.body;
+            });
+    };
+
+    /**
+     * @function createServiceProfile
+     * @description discovers all the service profiles in the Cisco UCS
+     */
+    UcsServiceProfileDiscoveryJob.prototype.createServiceProfile = function (root) {
+        var self = this;
+        var username = this.settings.username;
+        var password = this.settings.password;
+        var ucs = this.settings.ucs;
+        var baseurl = this.settings.protocol+"://"+this.settings.host+ ":"+this.settings.port;
+        var url=
+            baseurl + "/serviceProfile?host=" + ucs+ "&user="+ username + "&password=" + password;
+        return Promise.try(function () {
+            if (_.has(root, 'ServiceProfile')) {
+                return self.ucs.clientRequest(url)
+                    .then(function(res) {
+                        assert.object(res);
+                        return res.body;
+                    })
+                    .then(function(data) {
+                        assert.object(data);
+                        return data['ServiceProfile']['members']
+                    })
+                    .map(function(members){
+                        self.createNode(members, Constants.NodeTypes.Compute);
+                    });
+            }else{
+                logger.warning('No Logical Servers found');
+                return;
+            }
+        });
+    };
+
+    UcsServiceProfileDiscoveryJob.prototype.createNode = function(data, type) {
+
+        var self = this;
+        var config = Object.assign({}, self.settings);
+
+        assert.string(data.name);
+        assert.string(data.path);
+        var id;
+        id = [config.ucs, data.path];
```

Comment:
Please use the plural of `id` or change to other better name.

Because when I see the code `identifer: id` in line 174, I thought the `identifer` need be array, but you wrongly gave it a single id.  So I look back to check the node, and see the `id` is really an array.

My point is choosing a good variable name and don't bring surprise to other people who read the code.